dataprofiler/profilers/profile_builder.py

```diff
@@ -1199,10 +1200,10 @@ def __add__(self, other):
         merged_profile.hashed_row_dict.update(other.hashed_row_dict)
 
         # merge profiles
-        for profile_name in self._profile:
-            merged_profile._profile[profile_name] = (
-                self._profile[profile_name] + other._profile[profile_name]
-            )
+        for idx in range(len(self._profile)):
+            merged_profile._profile.append(self._profile[idx] +
+                                           other._profile[idx])
+        merged_profile._col_name_to_idx = copy.deepcopy(self._col_name_to_idx)
```

Comment:
prior to this, how have we validated that the two schemas are the same to allow for merging?

Reply:
the above only occurs if there are duplicate keys

Reply:
If schemas are slightly different, and no duplicate keys, I believe this would error because indexes could be different or non-existent for profiles that partially profiled different columns in a datasets.
maybe best to loop through the  `_col_name_to_idx` based on keys

Reply:
also in the case where we merge but not the same mapping, just copying the dict would not be an issue and instead would need to merge

Reply:
Or do we only allow merging of exact schema matches.

Reply:
^was thinking same thing, that schema could error when slightly different with this implementation

Reply:
In `_add_error_checks` the 2 profiles are required to have the same `_col_name_to_idx` mapping. I recall a conversation we had where you told me I had freedom to make schema requirements more strict in order to add this capability, is this no longer the case?